pkg/distributor/distributor.go

```diff
@@ -1871,15 +1877,28 @@ func newActiveSeriesResponse(hashCollisionCount prometheus.Counter) *activeSerie
 	}
 }
 
-func (r *activeSeriesResponse) add(series []*mimirpb.Metric) {
+var ErrResponseTooLarge = errors.New("response too large")
+
+const activeSeriesResponseMaxSize = 5 * 1024 * 1024
```

Comment:
I don't think this should be a constant, it should be a config. And it should be way higher, 5 MB is nothing IMO.